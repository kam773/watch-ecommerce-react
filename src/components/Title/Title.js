import React from "react";
import { Col, Row } from "reactstrap";

export default function Title({ name, title }) {
  return (
    <Row>
      <Col xs="10" className="mx-auto my-2 text-center text-title">
        <h1 className="text-capitalize font-weight-bold py-5">
          {name} <strong className="text-blue">{title}</strong>
        </h1>
      </Col>
    </Row>
  );
}
