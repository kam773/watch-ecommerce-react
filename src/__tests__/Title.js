import React from 'react';
import Title from '../components/Title/Title';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

describe("Title", () => {
    it('should render correctly', () => {
        const tree = renderer.create(<Title/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders heading tag", () => {
        const wrapper = shallow(<Title/>);
        expect(wrapper.find('h1')).toHaveLength(1);
    });
    it("renders strong tag", () => {
        const wrapper = shallow(<Title/>);
        expect(wrapper.find('strong')).toHaveLength(1);
    });
    it("should has class text-blue", () => {
        const wrapper = shallow(<Title/>);
        expect(wrapper.find('strong').hasClass('text-blue')).toEqual(true);
    });
    it("has a name", () => {
        const wrapper = mount(<h1>your</h1>)
        expect(wrapper.text()).toEqual('your');
    })
    it("has a title", () => {
        const wrapper = mount(<h1>cart</h1>)
        expect(wrapper.text()).toEqual('cart');
    })
})
