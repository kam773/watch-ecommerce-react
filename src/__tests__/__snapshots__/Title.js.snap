// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-10 mx-auto my-2 text-center text-title"
  >
    <h1
      className="text-capitalize font-weight-bold py-5"
    >
       
      <strong
        className="text-blue"
      />
    </h1>
  </div>
</div>
`;
